portalnet/src/storage.rs

```diff
@@ -744,6 +677,496 @@ impl PortalStorage {
+                                "Error looking up lc update period: {err:?}"
+                            ))
+                        })?
+                        .is_some();
+                    if is_period_available {
+                        return Ok(ShouldWeStoreContent::AlreadyStored);
+                    }
+                }
+                Ok(ShouldWeStoreContent::Store)
+            }
+            BeaconContentKey::LightClientFinalityUpdate(_) => {
+                // TODO: Check content key availability in cache
+                Ok(ShouldWeStoreContent::AlreadyStored)
+            }
+            BeaconContentKey::LightClientOptimisticUpdate(_) => {
+                // TODO: Check content key availability in cache
+                Ok(ShouldWeStoreContent::AlreadyStored)
+            }
+        }
+    }
+
+    fn radius(&self) -> Distance {
+        Distance::MAX
+    }
+}
+
+impl BeaconStorage {
+    pub fn new(config: PortalStorageConfig) -> Result<Self, ContentStoreError> {
+        let metrics = StorageMetricsReporter {
+            storage_metrics: PORTALNET_METRICS.storage(),
+            protocol: ProtocolId::Beacon.to_string(),
+        };
+        let storage = Self {
+            node_data_dir: config.node_data_dir,
+            sql_connection_pool: config.sql_connection_pool,
+            storage_capacity_in_bytes: config.storage_capacity_mb * BYTES_IN_MB_U64,
+            metrics,
+            network: ProtocolId::Beacon,
+        };
+
+        // Report current storage capacity.
+        storage
+            .metrics
+            .report_storage_capacity_bytes(storage.storage_capacity_in_bytes as f64);
+
+        // Report current total storage usage.
+        let total_storage_usage = storage.get_total_storage_usage_in_bytes_on_disk()?;
+        storage
+            .metrics
+            .report_total_storage_usage_bytes(total_storage_usage as f64);
+
+        // Report total storage used by network content.
+        let network_content_storage_usage =
+            storage.get_total_storage_usage_in_bytes_from_network()?;
+        storage
+            .metrics
+            .report_content_data_storage_bytes(network_content_storage_usage as f64);
+
+        Ok(storage)
+    }
+
+    fn db_insert(
+        &self,
+        content_id: &[u8; 32],
+        content_key: &String,
+        value: &Vec<u8>,
+    ) -> Result<(), ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        insert_value(conn, content_id, content_key, value, u8::from(self.network))
+    }
+
+    fn db_insert_lc_update(&self, period: &u64, value: &Vec<u8>) -> Result<(), ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        let value_size = value.len();
+
+        match conn.execute(
+            INSERT_LC_UPDATE_QUERY,
+            params![period, value, 0, value_size],
+        ) {
+            Ok(_) => Ok(()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    pub fn store(
```

Comment:
It feels like this (& any other shared methods inside this `impl BeaconStorage {}`) should be moved to the `ContentStore` trait... or maybe a new shared trait? I'm not sure, since IIRC `ContentStore` was primarily created to enable `MemoryContentStore` and use that inside tests. Maybe, we should create a new trait that will be shared amongst all subnetwork storage types. This could also include shared metrics reporting methods. Just thinking out loud.

Reply:
"store" is not a required method because it is wrapped by `put` which is in the `ContentStore` trait and this is what is used by tests and the overlay service. I'm using it in `BeaconStorage` because I try to follow a similar structure as `PortalStorage`, but this shouldn't be a hard requirement.

Anyway, I see some benefits of having additional traits but imo we will have more visibility here when we start to refactor the code.